cli/bpmetadata/types.go

```diff
@@ -18,27 +18,27 @@ type BlueprintMetadataSpec struct {
 
 // BlueprintInfo defines informational detail for the blueprint
 type BlueprintInfo struct {
-	Title         string               `yaml:"title"`
-	Source        *BlueprintRepoDetail `yaml:"source"`
+	Title         string
+	Source        *BlueprintRepoDetail
 	Version       string
-	ActuationTool *BlueprintActuationTool
-	Description   *BlueprintDescription
-	Icon          string
+	ActuationTool BlueprintActuationTool `json:"actuationTool" yaml:"actuationTool"`
+	Description   *BlueprintDescription  `json:",omitempty" yaml:",omitempty"`
+	Icon          string                 `json:",omitempty" yaml:",omitempty"`
 }
 
 // BlueprintContent defines the detail for blueprint related content such as
 // related documentation, diagrams, examples etc.
 type BlueprintContent struct {
 	Diagrams      []BlueprintDiagram
```

Comment:
I suspect we can omitempty quite a few of these.

Reply:
I can take another pass